driver/src/main/com/mongodb/reactivestreams/client/gridfs/GridFSBucket.java

```diff
@@ -232,51 +375,98 @@
     GridFSDownloadStream openDownloadStream(ObjectId id);
 
     /**
-     * Downloads the contents of the stored file specified by {@code id} and writes the contents to the {@code destination}
-     * AsyncOutputStream.
+     * Opens a AsyncInputStream from which the application can read the contents of the stored file specified by {@code id}.
      *
-     * @param id          the ObjectId of the file to be written to the destination stream
-     * @param destination the destination stream
-     * @return a publisher with a single element, representing the amount of data written
+     * @param id the custom id value of the file, to be put into a stream.
+     * @return the stream
      */
-    Publisher<Long> downloadToStream(ObjectId id, AsyncOutputStream destination);
+    GridFSDownloadStream openDownloadStream(BsonValue id);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the latest version of the stored file specified by the
+     * {@code filename}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @return the stream
+     */
+    GridFSDownloadStream openDownloadStream(String filename);
+
+    /**
+     * Opens a Stream from which the application can read the contents of the stored file specified by {@code filename} and the revision
+     * in {@code options}.
+     *
+     * @param filename the name of the file to be downloaded
+     * @param options  the download options
+     * @return the stream
+     */
```

Comment:
@since 1.7

Reply:
This just got moved to keep all the same named methods inline.

Reply:
Acknowledged